WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -457,6 +457,37 @@ public async Task SignTransactionAsync(TransactionSignaturesRequest request)
 			}
 		}
 
+		public async Task<ReissueCredentialResponse> ReissuanceAsync(ReissueCredentialRequest request)
+		{
+			using (await AsyncLock.LockAsync().ConfigureAwait(false))
+			{
+				if (!Rounds.TryGetValue(request.RoundId, out var round))
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.RoundNotFound, $"Round ({request.RoundId}) not found.");
+				}
+
+				if (request.RealAmountCredentialRequests.Requested.Count() != 2)
+				{
+					throw new NotImplementedException(); // TODO
+				}
+
+				if (round.Phase is not (Phase.ConnectionConfirmation or Phase.OutputRegistration))
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.WrongPhase, $"Round ({request.RoundId}): Wrong phase ({round.Phase}).");
```

Comment:
It is the same but after we validated the input and got the round instance I would forget about the request object and use the round instead.

```suggestion
					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.WrongPhase, $"Round ({round.Id}): Wrong phase ({round.Phase}).");
```

Reply:
ACK.